docker/Dockerfile.flux

```diff
@@ -29,21 +35,12 @@ LABEL maintainer="Weaveworks <help@weave.works>" \
       org.label-schema.vcs-url="git@github.com:weaveworks/flux" \
       org.label-schema.vendor="Weaveworks"
 
-ENTRYPOINT [ "/sbin/tini", "--", "fluxd" ]
+ENTRYPOINT [ "/usr/local/bin/tini", "--", "fluxd" ]
```

Comment:
Why so we need tini to run a go binary?

Reply:
Good question, was wondering this too, but as my goal was to simply port the current image, I did not change it.

Reply:
It at least used to be the case that it was not advisable to run a Go binary as pid 1, since it will not reap zombies.

Reply:
(i.e., using `tini` to run the Go binary is an easy way to make sure child processes are cleaned up.)

Reply:
@squaremo how come no one uses tini for containers? I haven't seen any CNCF project doing that, for example https://github.com/coredns/coredns/blob/master/Dockerfile#L12

Reply:
Either:

 - their binary handles zombies or doesn't create them (no child processes)
 - they are leaving it for users to solve
 - they are not aware of this problem

Reply:
Ah now I understand, we run git and kubectl as child processes while coredns and others don't. Thanks @squaremo